examples/mnist_tfaugment2d.py

```diff
@@ -0,0 +1,143 @@
+'''Trains a simple convnet on the MNIST dataset.
+
+Using Tensorflow internal augmentation APIs which is faster than ImageGenerator.
+
+# Benchmark
+
+1) Set horizontal_flip = True on MNIST_CNN model,
+Efficiency comparison: ImageGenerator and AugmentLayer (backend = Tensorflow-GPU)
+----------------------------------------------------------------------------
+Epoch     | ImageGenerator | ImageGenerator | AugmentLayer  | Augment Layer
+Number    | %Accuracy      | Performance    | %Accuracy     | Performance
+----------------------------------------------------------------------------
+1         | 95.18          | 18ms/step      | 94.90         | 132us/step
+2         | 96.73          |  9ms/step      | 96.57         |  71us/step
+4         | 97.58          |  9ms/step      | 97.23         |  70us/step
+8         | 98.15          |  9ms/step      | 97.72         |  70us/step
+12        | 98.27          |  9ms/step      | 98.10         |  71us/step
+---------------------------------------------------------------------------
+
+2) Set rotation = 120.0 on MNIST_CNN model,
+Efficiency comparison: ImageGenerator and AugmentLayer (backend = Tensorflow-GPU)
+----------------------------------------------------------------------------
+Epoch     | ImageGenerator | ImageGenerator | AugmentLayer  | Augment Layer
+Number    | %Accuracy      | Performance    | %Accuracy     | Performance
+----------------------------------------------------------------------------
+1         | 92.83          | 25ms/step      | 90.98         | 133us/step
+2         | 94.69          | 18ms/step      | 93.72         |  73us/step
+4         | 96.02          | 20ms/step      | 95.28         |  73us/step
+8         | 96.86          | 19ms/step      | 96.46         |  72us/step
+12        | 96.91          | 18ms/step      | 96.58         |  72us/step
+---------------------------------------------------------------------------
+
+
+Gets to 96.58% test accuracy after 12 epochs when rotation range is (-120, 120)
```

Comment:
Wait, really? This is unacceptably low for a MNIST model. Dense models get to above 98%, and a convnet should get above 99%.

Reply:
If using ImageGenerator to rotate (-120, 120) of MNIST dataset, the answer is also around 96% just after 12 epoches. I think the main reason is number 6 and 9 become similar after rotation while their outputs are different.